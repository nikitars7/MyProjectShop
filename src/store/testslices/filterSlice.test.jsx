import filterSliceReducer, {
  setSearchParams,
  setSortBy,
  setPage,
} from "../slices/filterSlice";
describe("filterSlice", () => {
  test("should return default state", () => {
    const result = filterSliceReducer(undefined, { type: "" });
    expect(result).toEqual({ searchParams: "", sortValue: "-name", page: 1 });
    expect(result).toMatchSnapshot();
  });
  test("should update search value", () => {
    const action = { type: setSearchParams.type, payload: "search..." };
    const result = filterSliceReducer(
      { searchParams: "", sortValue: "-name", page: 1 },
      action
    );
    const {searchParams} = result;
    expect(searchParams).toBe('search...');
    expect(result).toMatchSnapshot();
  });
  test("should update sort value", () => {
   const action = { type: setSortBy.type, payload: "price" };
   const result = filterSliceReducer(
     { searchParams: "", sortValue: "-name", page: 1 },
     action
   );
   const {sortValue} = result;
   expect(sortValue).toBe('price');
   expect(result).toMatchSnapshot();
 });
 test("should update page value", () => {
   const action = { type: setPage.type, payload: 2 };
   const result = filterSliceReducer(
     { searchParams: "", sortValue: "-name", page: 1 },
     action
   );
   const {page} = result;
   expect(page).toBe(2);
   expect(result).toMatchSnapshot();
 });
 afterEach(() => {
   jest.clearAllMocks();
 })
});
