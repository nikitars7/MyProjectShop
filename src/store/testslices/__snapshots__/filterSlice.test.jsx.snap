// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterSlice should return default state 1`] = `
Object {
  "page": 1,
  "searchParams": "",
  "sortValue": "-name",
}
`;

exports[`filterSlice should update page value 1`] = `
Object {
  "page": 2,
  "searchParams": "",
  "sortValue": "-name",
}
`;

exports[`filterSlice should update search value 1`] = `
Object {
  "page": 1,
  "searchParams": "search...",
  "sortValue": "-name",
}
`;

exports[`filterSlice should update sort value 1`] = `
Object {
  "page": 1,
  "searchParams": "",
  "sortValue": "price",
}
`;
